sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample08_EventProcessor.md

```diff
@@ -0,0 +1,230 @@
+
+Here's a simple processor that just counts the number of events received in a batch and then creates a checkpoint so they are not processed again:
+
+```C# Snippet:EventHubs_Sample08_CustomProcessor
+class CustomProcessor : AzureBlobStorageEventProcessor<EventProcessorPartition>
+{
+    public CustomProcessor(int eventBatchMaximumCount, string consumerGroup, string fullyQualifiedNamespace, string eventHubName, TokenCredential credential, BlobContainerClient storageContainer, EventProcessorOptions options = null) : base(eventBatchMaximumCount, consumerGroup, fullyQualifiedNamespace, eventHubName, credential, storageContainer, options)
+    {
+    }
+
+    protected async override Task OnProcessingEventBatchAsync(IEnumerable<EventData> events, EventProcessorPartition partition, CancellationToken cancellationToken)
+    {
+        Console.WriteLine($"Received batch of {events.Count()} events for partition {partition.PartitionId}");
+
+        if (events.Any())
+        {
+            await CheckpointAsync(partition, events.Last(), cancellationToken);
+        }
+    }
+
+    protected async override Task OnProcessingErrorAsync(Exception exception, EventProcessorPartition partition, string operationDescription, CancellationToken cancellationToken)
+    {
+        if (partition != null)
+        {
+            Console.Error.WriteLine($"Exception on partition {partition.PartitionId} while performing {operationDescription}: {exception.Message}");
+        }
+        else
+        {
+            Console.Error.WriteLine($"Exception while performing {operationDescription}: {exception.Message}");
+        }        
+    }    
+}
+```
+
+This should look familiar if you've used `EventProcessorClient` before, but there are a few small differences you need to be aware of:
+
+- The processor works in batches, so when you construct it you need to specify the maximum number of events you'll want to process at once, using the `eventBatchMaximumCount` parameter. The `IEnumerable<EventData>` that is passed `OnProcessingEventBatchAsync` to will never contain more items than this maximum count.
+- If there are fewer unprocessed messages than `eventBatchMaximumCount` the processor also contains a maximum amount of time it will wait for more events to show up. By default, this is 60 seconds, but can be controlled by setting the `MaximumWaitTime` property of the `EventProcessorOptions` parameter. If this is set to `null` it means that the processor will until it has received `eventBatchMaximumCount` events, otherwise it will deliver a batch that has fewer events in it.
```

Comment:
You may want to add spacing between the bullets.   I found the condensed text a bit difficult to read.